### PR TITLE
Rename (Shaped)Hand -> (Shaped)Hole

### PR DESCRIPTION
**Pull request by santiweight**
The new name is less likely to conflict with users'
libraries, which will often use the name Hand
for many different applications.

### CODE REVIEW

**Comment by santiweight** [comment]
I haven't been able to run docspec et al yet. They are still installing. I'll let ci help me out in the meantime...

File: src/Poker/Cards.hs [hunk 1]
```diff
@@ -198,83 +197,83 @@ instance Bounded Card where
...
   r2 <- enumFrom r1
   (s1, s2) <-
     if r1 == r2
       then [(s1, s2) | s1 <- enumerate, s2 <- drop 1 (enumFrom s1)]
       else liftM2 (,) enumerate enumerate
-  pure $ unsafeMkHand (Card r1 s1) (Card r2 s2)
+  pure $ unsafeMkHole (Card r1 s1) (Card r2 s2)
 
-instance Enum Hand where
+instance Enum Hole where
```
Comment (tonyday567) [inline src/Poker/Cards.hs:243]: Without digging into it, this instance doesn't look safe. Can we ditch all the enum instances?

I like the convention of allWhatever, and enumerate can be replaced with direct usage of these lists..

File: src/Poker/Cards.hs [hunk 2]
```diff
@@ -198,83 +197,83 @@ instance Bounded Card where
 allCards :: [Card]
 allCards = liftM2 Card allRanks allSuits
 
--- | 'Hand' represents a player's hole cards in a game of Texas Hold\'Em
-data Hand = MkHand !Card !Card
+-- | 'Hole' represents a player's hole cards in a game of Texas Hold\'Em
+data Hole = MkHole !Card !Card
```
Comment (tonyday567) [inline src/Poker/Cards.hs:201]: Consider UnsafeHold as an alternatve constructor name (or, alternatively, stop normalising and just go with Hole)

File: src/Poker/Cards.hs [hunk 3]
```diff
@@ -285,53 +284,53 @@ instance ParsePretty Hand where
...
--- >>> parsePretty @ShapedHand "24o"
+-- >>> parsePretty @ShapedHole "24o"
 -- Just (MkOffsuit Four Two)
--- >>> parsePretty @ShapedHand "24s"
+-- >>> parsePretty @ShapedHole "24s"
 -- Just (MkSuited Four Two)
 --
 -- TODO Make patterns uni-directional (don't expose constructors)
-data ShapedHand = MkPair !Rank | MkOffsuit !Rank !Rank | MkSuited !Rank !Rank
+data ShapedHole = MkPair !Rank | MkOffsuit !Rank !Rank | MkSuited !Rank !Rank
```
Comment (tonyday567) [inline src/Poker/Cards.hs:295]: consider #25

File: src/Poker/Cards.hs [hunk 4]
```diff
@@ -285,53 +284,53 @@ instance ParsePretty Hand where
...
 
-unsafeMkOffsuit :: HasCallStack => Rank -> Rank -> ShapedHand
+unsafeMkOffsuit :: HasCallStack => Rank -> Rank -> ShapedHole
 unsafeMkOffsuit r1 r2 =
   fromMaybe (terror $ "Cannot form offsuit hand from: " <> prettyText (r1, r2)) $
     mkOffsuit r1 r2
 
-listShapedHands :: [ShapedHand]
-listShapedHands = reverse $ do
+listShapedHoles :: [ShapedHole]
```
Comment (tonyday567) [inline src/Poker/Cards.hs:332]: allShapedHoles fits in with library naming conventions

File: src/Poker/Cards.hs [hunk 5]
```diff
@@ -340,57 +339,57 @@ listShapedHands = reverse $ do
...
+    pure $ unsafeMkHole (Card r1 s1) (Card r2 s2)
   Suited r1 r2 -> do
     s <- enumerate
-    pure $ unsafeMkHand (Card r1 s) (Card r2 s)
+    pure $ unsafeMkHole (Card r1 s) (Card r2 s)
 
 -- TODO needs tests
-handToShaped :: Hand -> ShapedHand
-handToShaped (Hand (Card r1 s1) (Card r2 s2))
+holeToShaped :: Hole -> ShapedHole
```
Comment (tonyday567) [inline src/Poker/Cards.hs:363]: holeToShapedHole

File: src/Poker/Range.hs [hunk 6]
```diff
@@ -79,10 +79,10 @@ getDecisionFreqRange ::
...
   Range $ Map.map (foldMap (\v -> Freq (bool 0 1 $ p v) 1)) m
 
-holdingRangeToShapedRange :: Monoid v => Range Hand v -> Range ShapedHand v
+holdingRangeToShapedRange :: Monoid v => Range Hole v -> Range ShapedHole v
 holdingRangeToShapedRange (Range r) =
-  Range $ Map.mapKeysWith (<>) handToShaped r
+  Range $ Map.mapKeysWith (<>) holeToShaped r
 
-addShaped :: Num a => a -> Hand -> Range ShapedHand a -> Range ShapedHand a
+addShaped :: Num a => a -> Hole -> Range ShapedHole a -> Range ShapedHole a
```
Comment (tonyday567) [inline src/Poker/Range.hs:86]: addHole because we also need:

addShapedHole :: a -> ShapedHole -> etc

**Review by tonyday567 — COMMENTED** [review]
I'm not one to "approve" or whatnot: move fast break stuff etc, but we should discuss what you might need from a review.

File: src/Poker/Cards.hs (hunk 1 above)
Comment (santiweight) [reply src/Poker/Cards.hs:243]: Agreed - I'll be removing this is future patches :)

File: src/Poker/Cards.hs (hunk 2 above)
Comment (santiweight) [reply src/Poker/Cards.hs:201]: Will follow up separately

**Pull request merged**